apps/system/lockscreen/js/widgets/clock/lockscreen_clock_widget.js

```diff
@@ -50,6 +52,44 @@
     this.logger.debug('Clock updated', now);
   };
 
+	LockScreenClockWidget.prototype.updateAlarm =
+	function() {
+    var self = this;
+		// returns alarm info from DataStore
+		navigator.getDataStores('alarms')
+			.then( function(stores){
+        stores[0].getLength().then(function(len){
+          if(len > 0) {
+            stores[0].get(1)
+              .then( function(data) {
+                if(data.data) {
+                  if(data.data.hour && data.data.minute) {
```

Comment:
If the `if(data.data)` only has one child branch, and it's `else` just do nothing, you can write it as

    if (!data.data) {
        return;
    }
    if (data.data.hour && data.data.minute) {
        // ....
    }

In this way we reduce the unnecessary sections just because you want to do nothing if there is no data after all.